### PR TITLE
Auto formatting

### LINKED ISSUE #1926: Aligning formatting accross files
Similar to [binding templates repository](https://github.com/w3c/wot-binding-templates/blob/main/.editorconfig), we can add an editorconfig file so that :

- The documents we need to edit the source code of is easier to do and we avoid unwanted whitespace changes.
- Some documents like JSON Schema can be also read by developers so we should provide a "nice looking" document to them.

I would exclude files that are auto-generated and also all the existing HTML files in the publication folder.

### PR DESCRIPTION
**Pull request by egekorkan**
fixes #1926

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# EditorConfig is awesome: https://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+[*]
+indent_style = space # useTabs in prettier
```
Comment (JKRhb) [inline .editorconfig:7]: Maybe the boolean value could be added here for clarification?

```suggestion
indent_style = space # useTabs=false in prettier
```

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+errata.html
+errata11.html
+
+# Automatically generated files
+ontology/*.html
+index.html
+validation/tm-json-schema-validation.json
+context/td-context-1.1.jsonld
+visualization/*.png
+visualization/*.svg
```
Comment (JKRhb) [inline .prettierignore:18]: A trailing could be added here as well:

```suggestion
visualization/*.svg

```

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# EditorConfig is awesome: https://EditorConfig.org
```
Comment (JKRhb) [inline .editorconfig:1]: Should we keep this line or should we maybe provide more context/documentation here?

**Review by JKRhb — COMMENTED** [review]
Thank you, @egekorkan! :) See three minor comments below.

File: .editorconfig (hunk 3 above)
Comment (egekorkan) [reply .editorconfig:1]:
```suggestion
# This file allows you to configure your editor using [EditorConfig](https://EditorConfig.org).
# Make sure to install the EditorConfig plugin if your editor doesn't come with built-in support.
```

Comment (egekorkan) [reply .editorconfig:1]: @JKRhb you can commit if that is what you had in mind :)

Comment (JKRhb) [reply .editorconfig:1]: Looks good, thank you, @egekorkan! :)

**Comment by egekorkan** [comment]
There are way too many conflicts with this PR. I am going to open a new one

**Pull request closed without merging**